### PR TITLE
Add reader options

### LINKED ISSUE #11: Add reader: :private option ?
Following poodr book by Sandy Metz, I tend to make internal dependencies private by default. To make my life easier as a developer,  I add `attr_reader` for these instance variables bellow a `private` keyword.

DSL could look like:
```ruby
class Thingy
  extend Dry::Initializer::Mixin

  option :my_internal_dependency, reader: :private
end
```

Plain ruby equivalent:
```ruby
class Thingy
  def initialize(my_internal_dependency:)
    @my_internal_dependency = my_internal_dependency
  end

  private
  
  attr_reader  :my_internal_dependency
end
```

Would you like a PR to add this feature?

### PR DESCRIPTION
**Pull request by jmgarnier**
Closes #11 

- [x] `reader: false`
- [x] `reader: :private`
- [x] `reader: :public` (default option)
- [x] `reader: :protected`

### CODE REVIEW

File: lib/dry/initializer/builder.rb [hunk 1]
```diff
@@ -76,10 +76,13 @@ def copy(&block)
     end
 
     def define_readers(mixin)
-      readers = @signature.select { |item| item.settings[:reader] != false }
-                          .map(&:name)
+      define_reader(mixin, :attr_reader, ->(item) { item.settings[:reader] != false })
+      define_reader(mixin, :private, ->(item) { item.settings[:reader] == :private })
```
Comment (jmgarnier) [inline lib/dry/initializer/builder.rb]: A quick ping to @nepalez before I move further. Given this is `dry-rb`, I thought I would DRY my contribution hoping it stays readable. Let me know if you are OK, cheers.

**Comment by jmgarnier** [comment]
note: tests pass but some rubies have compilation / download issues

File: lib/dry/initializer/builder.rb (hunk 1 above)
Comment (nepalez) [inline lib/dry/initializer/builder.rb]: I think we should add the same line for a :protected option as well

Comment (nepalez) [inline lib/dry/initializer/builder.rb]: @jmgarnier  You should rebase the branch to the master. I've fixed .travis.yml, so tests should pass

Comment (jmgarnier) [inline lib/dry/initializer/builder.rb]: @nepalez Thanks for your quick feedback. done

**Comment by jmgarnier** [comment]
@nepalez I am done with the scope of the issue. Now I am wondering if the `Builder` should validate `reader:` options and raise an `ArgumentError` when option is invalid.

**Comment by nepalez** [comment]
@jmgarnier Nice work! Thank you!

I don't think we should do that validation. With a current implementation a user has additional <undocumented> options, namely every value except for `false`, `:private`, `:protected` is treated as the default `:public`. For example, `reader: true` seems pretty reasonable, isn't it?

**Pull request merged**